podcasts/Notifications/NotificationsCoordinator.swift

```diff
@@ -0,0 +1,119 @@
+        case .onboardingUpNext:
+            return L10n.notificationsOnboardingUpnextBody
+        case .onboardingFilters:
+            return L10n.notificationsOnboardingFiltersBody
+        case .onboardingUpsell:
+            return L10n.notificationsOnboardingUpsellBody
+        }
+    }
+
+    var identifier: String {
+        return self.rawValue
+    }
+
+    var link: String {
+        switch self {
+        case .onboardingSignUp:
+            return "pktc://signup"
+        case .onboardingImport:
+            return "pktc://settings/import"
+        case .onboardingThemes:
+            return "pktc://settings/themes"
+        case .onboardingUpNext:
+            return "pktc://upnext/?location=tab"
+        case .onboardingFilters:
+            return "pktc://filters"
+        case .onboardingUpsell:
+            return "pktc://upsell"
+        }
+    }
+}
+
+class NotificationsCoordinator {
+
+    static var shared: NotificationsCoordinator = NotificationsCoordinator()
+
+    var timeIntervalStep: TimeInterval = 24.hours
+
+    private init() {
+
+    }
+
+    private let onboardingNotifications: [NotificationType] = [.onboardingSignUp, .onboardingImport, .onboardingUpNext, .onboardingFilters, .onboardingThemes, .onboardingUpsell]
+
+    func setupOnboardingNotifications() {
+        Settings.notificationsOnboardingTips = true
+        NotificationsHelper.shared.registerForPushNotifications { granted in
+            guard granted else { return }
+            var timeInterval: TimeInterval = self.timeIntervalStep
+            self.onboardingNotifications.forEach { notification in
+                self.scheduleNotification(notification, timeInterval: timeInterval)
+                timeInterval += self.timeIntervalStep
+            }
+        }
+    }
+
+    func cancelOnboardingNotifications() {
+        Settings.notificationsOnboardingTips = false
+        let notificationCenter = UNUserNotificationCenter.current()
+        notificationCenter.removePendingNotificationRequests(withIdentifiers: onboardingNotifications.map { $0.identifier })
+    }
+
+    func scheduleNotification(_ type: NotificationType, timeInterval: TimeInterval = 5.seconds) {
+        let content = UNMutableNotificationContent()
+        content.title = type.title
+        content.body = type.body
+        content.categoryIdentifier = NotificationsHelper.NotificationsCategory.deepLink.rawValue
+        content.userInfo = ["destination_url": type.link]
+
+        let trigger = UNTimeIntervalNotificationTrigger(timeInterval: timeInterval, repeats: false)
+
+        let request = UNNotificationRequest(identifier: type.identifier, content: content, trigger: trigger)
+
+        // Schedule the request with the system.
+        Task {
+            let notificationCenter = UNUserNotificationCenter.current()
+            do {
+                try await notificationCenter.add(request)
+            } catch {
+                // Handle errors that may occur during add.
```

Comment:
Should this include a log just in case?

Reply:
Added here: https://github.com/Automattic/pocket-casts-ios/pull/2943/commits/d4df9312ba6baed88e670c98a68fe95bdcba76d3